File: ncDados/dEfdRegistro.vb
Namespace nsEFD

  Public Class ColecaoRegistro
    Inherits List(Of dEfdRegistro)

    Public Sub Incluir(ByVal pCodigo As String, ByVal pQuantidade As String)
      Dim registro As dEfdRegistro = New dEfdRegistro()

      registro.codigo = pCodigo
      registro.quantidade = pQuantidade

      Me.Add(registro)
    End Sub
  End Class

  Public Class dEfdRegistro

    Private _codigo As String
        Private _quantidade As Decimal

        Public Property codigo() As String
      Get
        Return _codigo
      End Get
      Set(ByVal value As String)
        _codigo = value
      End Set
    End Property

        Public Property quantidade() As Decimal
            Get
                Return _quantidade
            End Get
            Set(ByVal value As Decimal)
                _quantidade = value
            End Set
        End Property

    End Class

End Namespace

File: ncModelos/dEfdRegistro.cs
using System;
using System.Collections.Generic;

namespace ncModelos
{
    public class ColecaoRegistro : List<dEfdRegistro>
    {
        public void Incluir(string pCodigo, string pQuantidade)
        {
            var registro = new dEfdRegistro();
            registro.codigo = pCodigo;
            registro.quantidade = decimal.Parse(pQuantidade);
            this.Add(registro);
        }
    }

    public class dEfdRegistro
    {
        private string _codigo;
        private decimal _quantidade;

        public string codigo
        {
            get => _codigo;
            set => _codigo = value;
        }

        public decimal quantidade
        {
            get => _quantidade;
            set => _quantidade = value;
        }
    }

}
